### ISSUE TITLE
Insight on the kotlin 2.0 situation (activity in feature branch)

### ISSUE THREAD

**Issue by ursusursus** [square/anvil#1071]
Hi,
me & many other patiently track this repo about updates on the kotlin 2.0 blocker for quite some time now.

I'm not pressuring anyone as this is free & opensource work, which I respect, but it would be helpful to have some insight from the team about the situation.

I read in the zac sweer's blog post that the hooks anvil needs were unexpectedly removed in the 2.0 compiler,
but then I noticed some activity in feature branch https://github.com/square/anvil/commits/joel.kotlin-2.0.21/

So, what is the situation? Did kotlin team put some hooks back in for us & therefore will anvil be able to support kotlin 2.0?

**Comment by steve-the-edwards** [comment]
We are actively working on sharing a roadmap wrt Kotlin 2.0 and K2 for Anvil. It's interrupted by US Thanksgiving holidays right now, but expect something soon.

**Comment by JoelWilcox** [comment]
The roadmap is now available [here](https://github.com/square/anvil/blob/main/docs/ROADMAP.md)

**Comment by mattinger** [comment]
So does the breaking of the dependency on dagger basically mean that Anvil is it's own separate DI library?  Right now it's a mechanism for assembling dagger components dynamically.

From what i can tell by reading the roadmap, that will no longer be the case as part of the K2 work. 

Or am i misreading something?

**Comment by JoelWilcox** [comment]
> So does the breaking of the dependency on dagger basically mean that Anvil is it's own separate DI library?

That's correct, when we get to that stage then Anvil will be independent of Dagger, although it will likely still share the same API initially.

> From what i can tell by reading the roadmap, that will no longer be the case as part of the K2 work.
>
> Or am i misreading something?

One clarification here is that they are sequential efforts. The work to get K2 supported in Anvil is happening currently but will not affect Anvil's relationship to Dagger. Once K2 support is finished, then the team will start on implementing its own Component generation and breaking the library dependency on Dagger. Let me know if that makes sense! Also happy to adjust the roadmap wording if you feel like it could be clearer about this :).

**Comment by JoelWilcox** [comment]
There's no specific target date at the moment since we're at a pretty early phase in the implementation. The best estimate I could give is that it's likely at least a few months away still.
